common/v2/services/Store/Contract/ContractProvider.tsx

```diff
@@ -20,6 +22,10 @@ export class ContractProvider extends Component {
       service.createContract(contractsData);
       this.getContracts();
     },
+    createContractWithId: (contractsData: ExtendedContract, id: string) => {
+      service.createContractWithId(contractsData, id);
+      this.getContracts();
+    },
```

Comment:
Agreed, in this case, we had to create the UUID manually so that after creating the contract, we are able to add the newly created contract UUID to the right network object.
For situations like this, it would be beneficial if the default create methods could return UUID after the creation of a new item (In this case contract).